app/client/src/utils/hooks/cyanWaterbodies.ts

```diff
@@ -0,0 +1,330 @@
+import Color from '@arcgis/core/Color';
+import FeatureLayer from '@arcgis/core/layers/FeatureLayer';
+import Graphic from '@arcgis/core/Graphic';
+import GroupLayer from '@arcgis/core/layers/GroupLayer';
+import MediaLayer from '@arcgis/core/layers/MediaLayer';
+import Polygon from '@arcgis/core/geometry/Polygon';
+import * as query from '@arcgis/core/rest/query';
+import SimpleFillSymbol from '@arcgis/core/symbols/SimpleFillSymbol';
+import SimpleRenderer from '@arcgis/core/renderers/SimpleRenderer';
+import { useCallback, useContext, useEffect, useRef, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+// contexts
+import { useFetchedDataDispatch } from 'contexts/FetchedData';
+import { LocationSearchContext } from 'contexts/locationSearch';
+import { useServicesContext } from 'contexts/LookupFiles';
+// utils
+import {
+  filterData,
+  handleFetchError,
+  updateFeatureLayer,
+  useBoundariesToggleLayer,
+  useLocalData,
+} from 'utils/hooks/boundariesToggleLayer';
+import { getPopupContent, getPopupTitle } from 'utils/mapFunctions';
+// types
+import type { FetchedDataAction, FetchState } from 'contexts/FetchedData';
+import type { Dispatch } from 'react';
+import type { NavigateFunction } from 'react-router-dom';
+import type {
+  CyanWaterbodyAttributes,
+  ServicesData,
+  ServicesState,
+} from 'types';
+import type { SublayerType } from 'utils/hooks/boundariesToggleLayer';
+
+/*
+## Hooks
+*/
+
+export function useCyanWaterbodies() {
+  const { data, status } = useLocalData(localFetchedDataKey);
+
+  return { cyanWaterbodies: data, cyanWaterbodiesStatus: status };
+}
+
+export function useCyanWaterbodiesLayers() {
+  // Build the base feature layer
+  const { getMapView } = useContext(LocationSearchContext);
+  const services = useServicesContext();
+  const navigate = useNavigate();
+
+  const buildBaseLayer = useCallback(
+    (type: SublayerType) => {
+      return buildLayer(navigate, services, type, getMapView);
+    },
+    [getMapView, navigate, services],
+  );
```

Comment:
```suggestion
export function useCyanWaterbodiesLayers() {
  // Build the base feature layer
  const { getMapView } = useContext(LocationSearchContext);
  const { getTitle, getTemplate } = useDynamicPopup();
  const services = useServicesContext();
  const navigate = useNavigate();

  const buildBaseLayer = useCallback(
    (type: SublayerType) => {
      return buildLayer(navigate, services, type, getMapView, getTitle, getTemplate);
    },
    [getMapView, getTemplate, getTitle, navigate, services],
  );
```

This is related to getting the Change to this location section in the surrounding features popups.